File: admin/src/main/java/com/beat/admin/promotion/api/response/BannerPresignedUrlFindResponse.java
package com.beat.admin.promotion.api.response;

import com.beat.admin.promotion.application.result.AdminPromotionPresignedUrlResults.BannerPresignedUrlResult;

public record BannerPresignedUrlFindResponse(
	String bannerPresignedUrl,
	BannerPresignedUploadResponse bannerPresignedUpload
) {
	private static BannerPresignedUrlFindResponse of(String bannerPresignedUrl, String bannerImageKey) {
		return new BannerPresignedUrlFindResponse(bannerPresignedUrl,
			new BannerPresignedUploadResponse(bannerPresignedUrl, bannerImageKey));
	}

	public static BannerPresignedUrlFindResponse from(BannerPresignedUrlResult result) {
		return BannerPresignedUrlFindResponse.of(result.bannerPresignedUrl(), result.bannerImageKey());
	}

	public record BannerPresignedUploadResponse(String uploadUrl, String imageKey) {
	}
}

File: admin/src/main/kotlin/com/beat/admin/promotion/api/response/BannerPresignedUrlFindResponse.kt
package com.beat.admin.promotion.api.response

import com.beat.admin.promotion.application.result.AdminPromotionPresignedUrlResults.BannerPresignedUrlResult

data class BannerPresignedUrlFindResponse(
    val bannerPresignedUrl: String,
    val bannerPresignedUpload: BannerPresignedUploadResponse,
) {
    constructor(result: BannerPresignedUrlResult) : this(
        bannerPresignedUrl = result.bannerPresignedUrl,
        bannerPresignedUpload = BannerPresignedUploadResponse(result.bannerPresignedUrl, result.bannerImageKey),
    )

    data class BannerPresignedUploadResponse(
        val uploadUrl: String,
        val imageKey: String,
    )
}